### PR TITLE
Update coverage tool

### PR DESCRIPTION
**Pull request by xhd2015**

### CODE REVIEW

File: cmd/xgo/coverage/main.go [hunk 1]
```diff
@@ -19,6 +19,10 @@ func Main(args []string) {
 		fmt.Print(strings.TrimPrefix(help, "\n"))
 		return
 	}
+	if cmd != "merge" && cmd != "compact" {
+		fmt.Fprintf(os.Stderr, "unrecognized cmd: %s\n", cmd)
+		return
```
Comment (ccoVeille) [inline cmd/xgo/coverage/main.go:24]: No os.Exit(1) here?

File: cmd/xgo/coverage/main.go [hunk 2]
```diff
@@ -19,6 +19,10 @@ func Main(args []string) {
 		fmt.Print(strings.TrimPrefix(help, "\n"))
 		return
 	}
+	if cmd != "merge" && cmd != "compact" {
+		fmt.Fprintf(os.Stderr, "unrecognized cmd: %s\n", cmd)
```
Comment (ccoVeille) [inline cmd/xgo/coverage/main.go:23]: No os. Exit(1) here?

File: cmd/xgo/coverage/main.go [hunk 3]
```diff
@@ -19,6 +19,10 @@ func Main(args []string) {
 		fmt.Print(strings.TrimPrefix(help, "\n"))
 		return
 	}
+	if cmd != "merge" && cmd != "compact" {
```
[the remark this answers is not in the document]
Comment (ccoVeille) [reply cmd/xgo/coverage/main.go:22]: I'm not even sure this code is needed as there is another switch after parsing flags

File: cmd/trace/main.go [hunk 4]
```diff
@@ -1,17 +1,12 @@
...
 )
 
 func main() {
 	args := os.Args[1:]
-	err := trace.Main(args)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
-	}
+	trace.Main(args)
```
Comment (ccoVeille) [inline cmd/trace/main.go:11]: May I suggest something?

Here and in previous code, you moved the `os.Exit(1)` inside the code, it's a bad pattern for me.

The code becomes almost impossible to test, because of the exit.

Maybe you made the change to be consistent with other part of code but I would have suggested to keep the error.

Why did you change ?

Also, this pattern leads to errors as os.Exit(1) can be forgotten

Comment (ccoVeille) [reply cmd/trace/main.go:11]: I can help in refactoring it, and suggest you something if you agree.

You may accept it or not, but I would like to show what I have in mind.

One question, do you plan to support other exit codes than 1?

File: cmd/xgo/coverage/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ccoVeille) [reply cmd/xgo/coverage/main.go:22]: This help command detection here should/should be combined with flagHelp, BTW

**Review by ccoVeille — COMMENTED** [review]
@xhd2015 my humble review to your project

File: cmd/xgo/coverage/main.go (hunk 2 above)
Comment (xhd2015) [reply cmd/xgo/coverage/main.go:23]: Good point

File: cmd/xgo/coverage/main.go [hunk 5]
```diff
@@ -82,35 +86,51 @@ func Main(args []string) {
...
 		}
-		covMode, cov := coverage.Parse(string(content))
+		_, cov := coverage.Parse(string(content))
 		covs = append(covs, cov)
-		if mode == "" {
-			mode = covMode
-		}
+		// if mode == "" {
+		// 	mode = covMode
+		// }
```
[the remark this answers is not in the document]
Comment (xhd2015) [reply cmd/xgo/coverage/main.go:122]: I forgot to delete them, they were intensionally commented out
